### PR TITLE
V3.6.0 jsb bridge

### PR DESCRIPTION
**Pull request by timlyeee**
Re: #https://github.com/cocos/3d-tasks/issues/10724

### Changelog

* modify finalize method to fit 3.6 behavior, otherwise an error would be triggered.
* release JsbBridge and JsbBridgeWrapper by dispatch close event.

-------

### Continuous Integration

This pull request:

* [ ] needs automatic test cases check.

  > Manual trigger with `@cocos-robot run test cases` afterward.

-------

### Compatibility Check

This pull request:

* [ ] changes public API, and have ensured backward compatibility with [deprecated features](https://github.com/cocos/cocos-engine/blob/v3.5.0/docs/contribution/deprecated-features.md).
* [ ] affects platform compatibility, e.g. system version, browser version, platform sdk version, platform toolchain, language version, hardware compatibility etc.
* [ ] affects file structure of the build package or build configuration which requires user project upgrade.
* [ ] **introduces breaking changes**, please list all changes, affected features and the scope of violation.

### CODE REVIEW

File: native/cocos/platform/apple/JsbBridgeWrapper.mm [hunk 1]
```diff
@@ -99,10 +101,14 @@ - (void)dispatchEventToScript:(NSString*)eventName {
     [jb sendToScript:eventName];
 }
 - (id)init {
-    self = [super init];
-    cbDictionnary = [NSMutableDictionary new];
-    jb = [JsbBridge sharedInstance];
-    [jb setCallback:cb];
+    if (self = [super init]) {
```
Comment (PatriceJiang) [inline native/cocos/platform/apple/JsbBridgeWrapper.mm:105]: When init failure, should we abort?

File: native/cocos/bindings/manual/JavaScriptJavaBridge.cpp [hunk 2]
```diff
@@ -686,7 +686,6 @@ se::Class *__jsb_ScriptNativeBridge_class = nullptr; // NOLINT
 static bool ScriptNativeBridge_finalize(se::State &s) { //NOLINT(readability-identifier-naming)
     auto *cobj = static_cast<ScriptNativeBridge *>(s.nativeThisObject());
     CC_ASSERT(cobj == ScriptNativeBridge::bridgeCxxInstance);
-    delete cobj;
     ScriptNativeBridge::bridgeCxxInstance = nullptr;
```
Comment (minggo) [inline native/cocos/bindings/manual/JavaScriptJavaBridge.cpp:688]: Does this method needed now?

File: native/cocos/bindings/manual/JavaScriptObjCBridge.mm [hunk 3]
```diff
@@ -408,7 +408,6 @@ bool register_javascript_objc_bridge(se::Object *obj) {
 static bool ScriptNativeBridge_finalize(se::State &s) { //NOLINT(readability-identifier-naming)
     auto *cobj = static_cast<ScriptNativeBridge *>(s.nativeThisObject());
     CC_ASSERT(cobj == ScriptNativeBridge::bridgeCxxInstance);
-    delete cobj;
     ScriptNativeBridge::bridgeCxxInstance = nullptr;
```
Comment (minggo) [inline native/cocos/bindings/manual/JavaScriptObjCBridge.mm:410]: The same as above.

File: native/cocos/platform/apple/JsbBridgeWrapper.mm (hunk 1 above)
Comment (timlyeee) [reply native/cocos/platform/apple/JsbBridgeWrapper.mm:105]: https://developer.apple.com/library/archive/documentation/General/Conceptual/CocoaEncyclopedia/Initialization/Initialization.html#//apple_ref/doc/uid/TP40010810-CH6-SW57
 should release and return nil here

**Review by minggo — APPROVED** [review]

**Pull request merged**